plugin-api/src/main/java/org/sonar/api/security/Authenticator.java

```diff
@@ -43,13 +45,31 @@ public abstract class Authenticator {
   public static final class Context {
     private String username;
     private String password;
+    /**
+     * @deprecated since 9.16 use {@link #httpRequest} instead
+     */
+    @Deprecated(since = "9.16", forRemoval = true)
     private HttpServletRequest request;
+    private HttpRequest httpRequest;
 
+    /**
+     * @deprecated since 9.16 use {@link #Context(String, String, HttpRequest)} instead
+     */
+    @Deprecated(since = "9.16", forRemoval = true)
     public Context(@Nullable String username, @Nullable String password, HttpServletRequest request) {
       requireNonNull(request);
       this.request = request;
       this.username = username;
       this.password = password;
+      this.httpRequest = new JavaxHttpRequest(request);
+    }
+
+    public Context(@Nullable String username, @Nullable String password, HttpRequest httpRequest) {
+      requireNonNull(httpRequest);
+      this.httpRequest = httpRequest;
+      this.username = username;
+      this.password = password;
+      this.request = (HttpServletRequest) httpRequest.getRawRequest();
```

Comment:
it'd probably be better for products to pass both `HttpRequest` and  `HttpServletRequest`.